nullaway/src/main/java/com/uber/nullaway/handlers/StreamNullabilityPropagator.java

```diff
@@ -348,7 +348,7 @@ public void onMatchMethodReference(
         if (elements.size() == 1) {
           // We only care for single method call chains (e.g. this.foo(), not this.f.bar())
           Element element = elements.get(0).getJavaElement();
-          if (!element.getKind().equals(ElementKind.METHOD)) {
+          if (!ElementKind.METHOD.equals(element.getKind())) {
```

Comment:
I would guess that this isn't the only place this bug will occur. I added this fix because it was the most minimal way to resolve the failing test, but I think we're better off with another solution involving filling out the custom `VariableElement` implementation with more data.

Using [Yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions) is defensive programming in general, but would result in a lot of code-churn to apply across the project.

Reply:
I find the NPE here to be disturbing, and I think we should aim for a more root-cause fix.  We actually check NullAway with NullAway itself now to try to prevent NPEs (the `:nullaway:buildWithNullAway` Gradle task).  It doesn't report an error here since the return of `com.uber.nullaway.dataflow.AccessPathElement#getJavaElement` is not marked `@Nullable`.  But, this means somehow an `AccessPathElement` is being created with `null` Java element, which is bad.  This is most likely due to `null` being returned from some unannotated API, but I'd like to make sure.  Could you possibly add null checks in [these constructors](https://github.com/uber/NullAway/blob/e9060eea7df27fbe2fe6d1dabd439ca0b5b2de74/nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathElement.java#L20-L28) to try to track which code is constructing this malformed `AccessPathElement`?  I think that will lead us to a better fix.

Reply:
Yes, I agree. I added a bit more information about the root cause in the linked github issue #676, the issue is here: https://github.com/uber/NullAway/blob/fcf045b27a27c4eafc739ad44f81af4fe2bdebd4/nullaway/src/main/java/com/uber/nullaway/handlers/OptionalEmptinessHandler.java#L233-L304

By extending a type where nulls are not expected and adding a `@Nullable` annotation, we violate Liskov substitution.

I think we want to hydrate that `VariableElement` with additional details to more closely approximate a field element, however that will come at a runtime performance cost (though I expect a minimal one).

Reply:
Ah yes, I see it now.  Sorry, didn't understand your comment before or look closely enough.  Let me think a bit more about whether we can easily flesh out this anonymous class.  Otherwise we can go with your local fix for now.

Reply:
I've created #678 to illustrate this alternative